src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -1271,9 +1298,20 @@ export default class DefaultDeviceController implements DeviceControllerBasedMed
       return stream.id === (device as MediaStream).id;
     }
 
-    const settings = stream.getTracks()[0].getSettings();
+    const settings = this.getMediaTrackSettings(stream);
     // If a device does not specify deviceId, we have to assume the stream is not reusable.
-    return settings.deviceId === this.getIntrinsicDeviceIdStr(device);
+    if (!settings.deviceId) {
+      return false;
+    }
+    const deviceIds = DefaultDeviceController.getIntrinsicDeviceId(device);
+    if (Array.isArray(deviceIds)) {
+      return deviceIds.includes(settings.deviceId);
+    }
```

Comment:
Probably not unless it's a single-element array — this check determines whether we can apply new constraints to the input, and that won't necessarily be true. It's safer to just return `false` here for a multi-element array.

Reply:
Update to return false for Array in both hasSameGroupId and isMediaStreamReusableByDeviceId